### PR TITLE
fix: Fixing local node

### PR DESCRIPTION
**Pull request by mm-zk**
## What ❔

* Fixing docker build and scripts for the local node/setup.

## Why ❔

* This will allow local node to work again - which is crucial before the bridgehub udpate.
* This PR is only fixing it in the 'main' branch - there will be a followup PR to the bridgehub branch coming later.

### CODE REVIEW

File: docker/local-node/entrypoint.sh [hunk 1]
```diff
@@ -1,46 +1,96 @@
...
+    update_config "/etc/env/base/database.toml" "state_keeper_db_path" "/var/lib/zksync/data/state_keeper"
+    update_config "/etc/env/base/database.toml" "backup_path" "/var/lib/zksync/data/backups"
 
-    # Switch zksolc compiler source from docker to binary
-    sed -i "s!'docker'!'binary'!" /contracts/l2-contracts/hardhat.config.ts
+    zk config compile
+  
+
+    zk db drop
+    zk db setup
```
Comment (ly0va) [inline docker/local-node/entrypoint.sh:73]: nit: equivalent to `zk db reset`

**Review by ly0va — DISMISSED** [review]

File: docker/local-node/entrypoint.sh (hunk 1 above)
Comment (mm-zk) [reply docker/local-node/entrypoint.sh]: done.

**Review by ly0va — APPROVED** [review]

**Review by RomanBrodetski — APPROVED** [review]
don't see anything suspicious

File: docker/local-node/Dockerfile [hunk 2]
```diff
@@ -1,35 +1,37 @@
...
-# Copy compiler (both solc and zksolc) binaries
-# Obtain `solc` 0.8.12.
-RUN wget https://github.com/ethereum/solc-bin/raw/gh-pages/linux-amd64/solc-linux-amd64-v0.8.12%2Bcommit.f00d7308 \
-    && mv solc-linux-amd64-v0.8.12+commit.f00d7308 /usr/bin/solc \
-    && chmod +x /usr/bin/solc
+ENV RUSTUP_HOME=/usr/local/rustup \
+    CARGO_HOME=/usr/local/cargo \
+    PATH=/usr/local/cargo/bin:$PATH
+
+RUN curl https://sh.rustup.rs -sSf | bash -s -- -y && \
```
Comment (yorik) [inline docker/local-node/Dockerfile:24]: It would be better to do `apt-get install rust`

**Review by yorik — APPROVED** [review]

File: docker/local-node/Dockerfile (hunk 2 above)
Comment (mm-zk) [reply docker/local-node/Dockerfile:24]: apt-get install rust fails - saying that there is no such package...

If you have other suggestion, I can fix it in the next PR.

**Pull request merged**